src/osd/ReplicatedPG.cc

```diff
@@ -2820,7 +2820,9 @@ ReplicatedPG::RepGather *ReplicatedPG::trim_object(const hobject_t &coid)
     coi.version = ctx->at_version;
     bl.clear();
     ::encode(coi, bl);
-    setattr_maybe_cache(ctx->obc, ctx, t, OI_ATTR, bl);
+    map<string, bufferlist> attrs;
+    attrs[OI_ATTR] = bl;
+    setattrs_maybe_cache(ctx->obc, ctx, t, attrs);
```

Comment:
Does it help to convert these cases where there is only 1 attr being set?  I see that it's an improvement below when you set 2 of them, but htere it just means constructing a mostly-useless STL map<>.